core/templates/dev/head/domain/utilities/ProbabilityService.js

```diff
@@ -0,0 +1,30 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for performing probabilistic operations.
+ */
+
+oppia.factory('ProbabilityService', [function() {
+  return {
+    /**
+     * @param {float} ratio - The probability for recording an issue, a number
```

Comment:
No idea why this is called "ratio", btw. A ratio is a comparison between two numbers. You seem to be describing a number between 0 and 1, which might be better conceptualized as a "successThreshold" or similar.

Reply:
Redundant now